priv/apps/itest/lib/in_flight_exit_client.ex

```diff
@@ -0,0 +1,262 @@
+
+    signed_txbytes = rlp_list |> ExRLP.encode() |> Encoding.to_hex()
+
+    %{se | signed_txbytes: signed_txbytes}
+  end
+
+  defp get_exit_game_contract_address(se) do
+    data = ABI.encode("exitGames(uint256)", [PaymentType.simple_payment_transaction()])
+    {:ok, result} = Ethereumex.HttpClient.eth_call(%{to: Itest.Account.plasma_framework(), data: Encoding.to_hex(data)})
+
+    exit_game_contract_address =
+      result
+      |> Encoding.to_binary()
+      |> ABI.TypeDecoder.decode([:address])
+      |> hd()
+      |> Encoding.to_hex()
+
+    %{se | exit_game_contract_address: exit_game_contract_address}
+  end
+
+  defp do_in_flight_exit(
+         %{address: address, exit_data: exit_data, exit_game_contract_address: exit_game_contract_address} = se
+       ) do
+    in_flight_tx = exit_data["in_flight_tx"] |> Encoding.to_binary()
+    in_flight_tx_sigs = Enum.map(exit_data["in_flight_tx_sigs"], &Encoding.to_binary(&1))
+    input_txs = Enum.map(exit_data["input_txs"], &Encoding.to_binary(&1))
+    input_txs_inclusion_proofs = Enum.map(exit_data["input_txs_inclusion_proofs"], &Encoding.to_binary(&1))
+    input_utxos_pos = Enum.map(exit_data["input_utxos_pos"], &:binary.encode_unsigned(&1))
+
+    # NOTE: hardcoded for now, we're talking to a particular exit game so this is fixed
+    optional_bytes_array = List.duplicate("", Enum.count(input_txs))
+
+    values = [
+      {in_flight_tx, input_txs, input_utxos_pos, optional_bytes_array, input_txs_inclusion_proofs, optional_bytes_array,
+       in_flight_tx_sigs, optional_bytes_array}
+    ]
+
+    data =
+      ABI.encode(
+        "startInFlightExit((bytes,bytes[],uint256[],bytes[],bytes[],bytes[],bytes[],bytes[]))",
+        values
+      )
+
+    txmap = %{
+      from: address,
+      to: exit_game_contract_address,
+      value: Encoding.to_hex(@ife_bond_size),
+      data: Encoding.to_hex(data),
+      gas: Encoding.to_hex(@ife_gas * 2),
+      gasPrice: Encoding.to_hex(@ife_gas_price)
+    }
+
+    {:ok, receipt_hash} = Ethereumex.HttpClient.eth_send_transaction(txmap)
+    wait_on_receipt_confirmed(receipt_hash, @retry_count)
+
+    %{se | start_in_flight_exit: receipt_hash}
+  end
+
+  defp get_in_flight_exit_id(%{exit_data: exit_data, exit_game_contract_address: exit_game_contract_address} = se) do
+    txbytes = exit_data["in_flight_tx"] |> Encoding.to_binary()
+    data = ABI.encode("getInFlightExitId(bytes)", [txbytes])
+    {:ok, result} = Ethereumex.HttpClient.eth_call(%{to: exit_game_contract_address, data: Encoding.to_hex(data)})
+
+    exit_id =
+      result
+      |> Encoding.to_binary()
+      |> ABI.TypeDecoder.decode([{:uint, 128}])
+      |> hd()
+
+    %{se | exit_id: exit_id}
+  end
+
+  defp piggyback_input(
+         %{
+           address: address,
+           exit_data: exit_data,
+           exit_game_contract_address: exit_game_contract_address
+         } = se
+       ) do
+    Process.sleep(10_000)
```

Comment:
why the sleep? and why 10 seconds?

Reply:
Maybe something like `(MIN_EXIT_PERIOD / 2) - 1_000`  would read better? I think the goal of the sleep is to wait long enough for the in-flight exit to be received, but not so long that we miss our window of opportunity (minimum exit period / 2).

Reply:
yeah, also... can we pull this out of the contract somehow? use poller?

Reply:
This is some tricky business... I don't see any other way than waiting X seconds (depending on the `MIN_EXIT_PERIOD` as suggested by @chrishunt). Any idea how you you would use poller in that case @InoMurko?

Reply:
waiting is fine! but if we could pull the "amount of waiting we need to do" out of the contracts that would make the code far more understandable and adaptable!

Reply:
I'm not screaming, just politely saying very loud! 🥇